### PR TITLE
feat: introduce `RenewAdminState` action

### PR DESCRIPTION
**Pull request by moreal**
## Background

The `AdminState` will be soon expired after the 3153600 block number. When the `AdminState` was expired, some actions like `PatchTableSheet` and `CreatePendingAccount` will not work. But the Proof-of-Stake or other system is not built yet.

## Description

So this pull request introduces `RenewAdminState` action which renew `AdminState.ValidUntil` field. It requires Admin permission so it should be executed before the block 3153600.

### CODE REVIEW

**Comment by longfin** [comment]
It would be better we have CLI commands to create and sign this action via 9c-tools.

**Comment by moreal** [comment]
@longfin Thanks for your suggestion.

I changed more after you looked:

- Introduced `9c-tools tx renew-admin-state <NEW-VALID-UNTIL>` command. (ref. https://github.com/planetarium/lib9c/pull/805#issuecomment-1005707992)
- `RenewAdminState` became to renew `AdminState`'s `ValidUntil` field even the `AdminState` is already expired. (ref. https://github.com/planetarium/lib9c/pull/805/commits/4b7cc0aa8dfd58ff3ba12b7ecddfd0acc051be39)

**Comment by area363** [comment]
@moreal Could you redirect this PR to https://github.com/planetarium/NineChronicles.Headless/tree/rc-v100098 so that it can be included in the new rc? 🙏

**Comment by moreal** [comment]
> @moreal Could you redirect this PR to https://github.com/planetarium/NineChronicles.Headless/tree/rc-v100098 so that it can be included in the new rc? 🙏

Thank you for letting me know it. I redirect this PR to the rc-v100098 branch and added this PR to the v100098 project board. 🙏🏻 ✅

**Review by longfin — CHANGES_REQUESTED** [review]
We need to add [a type dispatcher in `Tx.Sign()` method](https://github.com/planetarium/lib9c/blob/1700da1dd178d0079d8d33b2199fc23e8139f7ad/.Lib9c.Tools/SubCommand/Tx.cs#L81-L91) too.

File: .Lib9c.Tools/SubCommand/Tx.cs [hunk 1]
```diff
@@ -250,5 +251,16 @@ public void MigrationActivatedAccountsState()
...
+
+        [Command(Description = "Create RenewAdminState action and dump it.")]
+        public void RenewAdminState(
+            [Argument("NEW-VALID-UNTIL")]
+            long newValidUntil
+        )
+        {
+            PolymorphicAction<ActionBase> action = new RenewAdminState(newValidUntil);
+            byte[] raw = _codec.Encode(action.PlainValue);
+            Console.WriteLine(ByteUtil.Hex(raw));
```
Comment (longfin) [inline .Lib9c.Tools/SubCommand/Tx.cs:263]: We need to specify `nameof(RedeemAdminState)` as action type.

Comment (moreal) [reply .Lib9c.Tools/SubCommand/Tx.cs:263]: I had misunderstood their logic 😞 I'll patch it now.

**Review by area363 — APPROVED** [review]

File: .Lib9c.Tools/SubCommand/Tx.cs (hunk 1 above)
Comment (moreal) [reply .Lib9c.Tools/SubCommand/Tx.cs:268]: I amended it in f3ecdde. Please review again! @longfin

**Review by longfin — APPROVED** [review]

**Pull request merged**